test/test.js

```diff
@@ -13,8 +13,10 @@ var Primus = require('primus')
 function client(srv, primus, options){
   var addr = srv.address() || {}
     , address = addr.address || 'localhost'
-    , port = addr.port || 8080
-    , url = 'http://' + address + ':' + port;
+    , port = addr.port || 8080;
+  // Node 0.12 IPv6 fix
+  if (address === '::') address = 'localhost';
+  var url = 'http://' + address + ':' + port;
```

Comment:
I don't think it makes any sense to use anything but `localhost`; we have no need to support arbitrary IPv6 addresses because we intentionally bind the server to `localhost` by instantiating it without an address.

I believe your first suggestion is the easiest to grok, and we won't have problems with it.